connectors/email/src/main/java/io/camunda/connector/email/authentication/Authentication.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.email.authentication;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import io.camunda.connector.generator.java.annotation.TemplateDiscriminatorProperty;
+import java.util.Optional;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type", defaultImpl = NoAuthentication.class)
+@JsonSubTypes({
+  @JsonSubTypes.Type(value = SimpleAuthentication.class, name = "simple"),
+  @JsonSubTypes.Type(value = NoAuthentication.class, name = "oauth"),
+})
+@TemplateDiscriminatorProperty(
+    label = "Authentication",
+    group = "authentication",
+    name = "type",
+    defaultValue = "simple",
+    description = "")
+public sealed interface Authentication permits NoAuthentication, SimpleAuthentication {
+
+  String getSender();
+
+  Optional<String> getUser();
+
+  Optional<String> getSecret();
```

Comment:
What about `username´ and `password` instead of `User` and `Secret`?
Also wondering why they are wrapped in an `Optional`? 

Is `sender` really part of the authentication?

Reply:
In the future, we might add a Oauth2 authentication as some servers does not allow user/password authentication.
But those protocols works in the same ways under the hood, so secret is more generic to retrieve either the password or the secret token.

Sender might be the wrong name, that is the email we want to connect to:

`sender` is a bit more complicated, I agree it should probably be only on the SMTP send email action. But that would bring redundancy, the user would authenticate to `example@camunda.com` and he would have to enter a `from` variable on the action.

and we can not use the getUser function because in the future with Oauth2 authentication, the user is not the email address. This might be the easier method cause as of now, `getSender` and `getUser` returns the same value, but that might not be the case in another authentication implementation. I also agree the name is not the best, something like `emailAddress` might be better.

Let me know